### PR TITLE
Last Respects effect + Fixed Supreme Overlord

### LINKED ISSUE #3035: Supreme Overlord's effect is wrong
Its current implementation has an "initial x1.2 boost", but according to [Anubis](https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-20#post-9421520), Supreme Overlord only gets an additional x0.1 boost per fainted battler.

EDIT: Additionally, it's supposed to use a faint counter that's independent from this ability, and it's also used by Last Respects

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
SV introduced a variable for each side to count the amount of fainted Pokémon on that side. It caps at 100, and both Last Respects and Supreme Overlord use it.
However, but Supreme Overlord's effect caps at 5 and it's kept the same based on the value that the side counter was when the battler entered battle, meaning that a partner fainting while the ability's holder is also in won't affect its multiplier.

## Features this PR does NOT handle
Automatic test for checking the Last Respects 100 cap

## Issue(s) that this PR fixes
Fixes #3035 

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -4158,23 +4157,14 @@ bool8 ChangeTypeBasedOnTerrain(u8 battlerId)
...
-    for (i = 0; i < PARTY_SIZE; i++)
-    {
-        if (GetMonData(&party[i], MON_DATA_SPECIES) != SPECIES_NONE
-         && !GetMonData(&party[i], MON_DATA_IS_EGG)
-         && GetMonData(&party[i], MON_DATA_HP) == 0)
-            modifier += (!appliedFirstBoost) ? UQ_4_12(0.2) : UQ_4_12(0.1);
-        appliedFirstBoost = TRUE;
-    }
+    for (i = 0; i < gBattleStruct->supremeOverlordCounter[battlerId]; i++)
+        modifier += UQ_4_12(0.1);
```
Comment (LOuroboros) [inline src/battle_util.c:4167]: Won't this cap at x1.5? Supreme Overlord caps at x1.6, and as explained in the comment that you removed, the first ally that faints provides a x0.2 boost, while each other member of the party add a x0.1 to the multiplier when they faint.

Comment (AsparagusEduardo) [reply src/battle_util.c:4167]: > Won't this cap at x1.5?

[Yes](https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9421520).

**Comment by ZnogyroP** [comment]
Two small issues with the current implementation - first, Supreme Overlord's popup appears even if no allies have fainted in the current battle. Second, its name gets slightly cut off in the ability popup - it's more noticeable when it activates for an enemy, but it happens on both sides.

File: include/battle.h [hunk 2]
```diff
@@ -678,6 +677,8 @@ struct BattleStruct
     bool8 trainerSlideBeforeFirstTurnMsgDone;
     u32 aiDelayTimer; // Counts number of frames AI takes to choose an action.
     u32 aiDelayFrames; // Number of frames it took to choose an action.
+    u8 faintedMonCount[NUM_BATTLE_SIDES];
```
Comment (kleeenexfeu) [inline include/battle.h:680]: You can use these fields : https://github.com/rh-hideout/pokeemerald-expansion/blob/master/include/battle.h#L369-L373

**Pull request closed without merging**